modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ddl/DdlCommandHandler.java

```diff
@@ -134,115 +127,102 @@ public DdlCommandHandler(
-        return catalogManager.execute(cmd)
-                .handle(handleModificationResult(cmd.ifTableExists(), TableNotFoundValidationException.class));
+        return catalogManager.execute(cmd).handle(handleModificationResult());
     }
 
     /** Handles drop column command. */
     private CompletableFuture<@Nullable Long> handleAlterDropColumn(AlterTableDropColumnCommand cmd) {
-        return catalogManager.execute(cmd)
-                .handle(handleModificationResult(cmd.ifTableExists(), TableNotFoundValidationException.class));
+        return catalogManager.execute(cmd).handle(handleModificationResult());
     }
 
     /** Handles drop column command. */
     private CompletableFuture<@Nullable Long> handleAlterColumn(AlterTableAlterColumnCommand cmd) {
-        return catalogManager.execute(cmd)
-                .handle(handleModificationResult(cmd.ifTableExists(), TableNotFoundValidationException.class));
+        return catalogManager.execute(cmd).handle(handleModificationResult());
     }
 
     /** Handles create index command. */
     private CompletableFuture<@Nullable Long> handleCreateIndex(AbstractCreateIndexCommand cmd) {
         return catalogManager.execute(cmd)
-                .thenCompose(catalogVersion -> inBusyLock(busyLock, () -> waitTillIndexBecomesAvailableOrRemoved(cmd, catalogVersion)))
-                .handle(handleModificationResult(cmd.ifNotExists(), IndexExistsValidationException.class));
+                .thenCompose(applyResult -> inBusyLock(busyLock, () -> waitTillIndexBecomesAvailableOrRemoved(cmd, applyResult)))
+                .handle(handleModificationResult());
     }
 
     /** Handles drop index command. */
     private CompletableFuture<@Nullable Long> handleDropIndex(DropIndexCommand cmd) {
-        return catalogManager.execute(cmd)
-                .handle(handleModificationResult(cmd.ifExists(), IndexNotFoundValidationException.class));
+        return catalogManager.execute(cmd).handle(handleModificationResult());
     }
 
     /** Handles create schema command. */
     private CompletableFuture<Long> handleCreateSchema(CreateSchemaCommand cmd) {
-        return catalogManager.execute(cmd)
-                .handle(handleModificationResult(cmd.ifNotExists(), SchemaExistsException.class));
+        return catalogManager.execute(cmd).handle(handleModificationResult());
     }
 
     /** Handles drop schema command. */
     private CompletableFuture<Long> handleDropSchema(DropSchemaCommand cmd) {
-        return catalogManager.execute(cmd)
-                .handle(handleModificationResult(cmd.ifExists(), SchemaNotFoundException.class));
+        return catalogManager.execute(cmd).handle(handleModificationResult());
     }
 
-    private BiFunction<Integer, Throwable, @Nullable Long> handleModificationResult(boolean ignoreExpectedError, Class<?> expErrCls) {
-        return (ver, err) -> {
+    private BiFunction<CatalogApplyResult, Throwable, @Nullable Long> handleModificationResult() {
```

Comment:
no need to return BiFuction anymore, the whole method may become BiFunction

Reply:
fixed